src/send-bsd.h

```diff
@@ -34,4 +35,48 @@ int send_packet(sock_t sock, void *buf, int len, UNUSED uint32_t idx)
 	return write(sock.sock, buf, len);
 }
 
+// Since BSD doesn't have the sendmmsg syscall leveraged in send-linux.c, this just wraps the single send_packet call.
+// However, the behavior in sendmmsg is to send as many packets as possible until one fails, and then return the number of sent packets.
+// Following the same pattern for consistency
+// Returns - number of packets sent
+// Returns last error code if no packets could be sent successfully
+int send_batch(sock_t sock, batch_t* batch, int retries) {
+	int packets_sent = 0;
+	int retry_ct = 0;
```

Comment:
This looks like it can be defined in the loop, it's not used outside of it.

Reply:
The compiler is realistically going to take care of this — it's not really worth worrying about.